src/net/http.rs

```diff
@@ -3747,36 +3775,135 @@ impl HttpResponseType {
         ))
     }
 
-    fn parse_post_mempool_query<R: Read>(
-        _protocol: &mut StacksHttp,
-        request_version: HttpVersion,
-        preamble: &HttpResponsePreamble,
+    /// Read the trailing page ID from a transaction stream
+    fn parse_mempool_query_page_id<R: Read>(
+        pos: usize,
+        retry_reader: &mut RetryReader<'_, R>,
+    ) -> Result<Option<Txid>, net_error> {
+        // possibly end-of-transactions, in which case, the last 32 bytes should be
+        // a page ID.  Expect end-of-stream after this.
+        retry_reader.set_position(pos);
+        let next_page: Txid = match read_next(retry_reader) {
+            Ok(txid) => txid,
+            Err(e) => match e {
+                codec_error::ReadError(ref ioe) => match ioe.kind() {
+                    io::ErrorKind::UnexpectedEof => {
+                        if pos == retry_reader.position() {
+                            // this is fine -- the node didn't get another page
+                            return Ok(None);
+                        } else {
+                            // partial data -- corrupt stream
+                            test_debug!("Unexpected EOF: {} != {}", pos, retry_reader.position());
+                            return Err(e.into());
+                        }
+                    }
+                    _ => {
+                        return Err(e.into());
+                    }
+                },
+                e => {
+                    return Err(e.into());
+                }
+            },
+        };
+
+        test_debug!("Read page_id {:?}", &next_page);
+        Ok(Some(next_page))
+    }
+
+    /// Decode a transaction stream, returned from /v2/mempool/query.
+    /// The wire format is a list of transactions (no SIP-003 length prefix), followed by an
+    /// optional 32-byte page ID.  Obtain both the transactions and page ID, if it exists.
+    pub fn decode_tx_stream<R: Read>(
```

Comment:
In this method, it looks like none of the `ExpectedEndOfStream` error cases have coverage -- can these be covered by unit tests?

Reply:
Yup; it's working locally